server/src/main/java/org/elasticsearch/search/aggregations/metrics/scripted/ScriptedMetricAggregationBuilder.java

```diff
@@ -222,8 +222,11 @@ protected ScriptedMetricAggregatorFactory doBuild(SearchContext context, Aggrega
                 ScriptedMetricAggContexts.CombineScript.CONTEXT);
             combineScriptParams = combineScript.getParams();
         } else {
-            compiledCombineScript = (p, a) -> null;
-            combineScriptParams = Collections.emptyMap();
+            throw new IllegalArgumentException("[combineScript] must not be null: [" + name + "]");
+        }
+
+        if(reduceScript == null) {
```

Comment:
Could we do these null checks at the start of the method before we compile any scripts?

Reply:
Will fix.